pyidr/study_parser.py

```diff
@@ -308,35 +324,143 @@ def check(obj):
         gateway.close()
 
 
+class BasePrinter(object):
+
+    def __init__(self, parser):
+        self.parser = parser
+
+
+class JsonPrinter(BasePrinter):
+
+    def __init__(self, parser):
+        BasePrinter.__init__(self, parser)
+        self.objects = []
+
+    def consume(self, obj):
+        m = {
+            "source": self.parser._study_file,
+            "name": obj.name,
+            "description": obj.description,
+            "map": dict((v[0], v[1]) for v in obj.map),
+        }
+        if hasattr(obj, "files"):
+            m["files"] = obj.files
+        self.objects.append(m)
+
+    def finish(self):
+        print json.dumps(self.objects, indent=4, sort_keys=True)
+
+
+class HCAPrinter(JsonPrinter):
+
+    def finish(self):
+        rv = []
+        for idx, obj in enumerate(self.objects):
+
+            # Setup
+            hca = {}
+            rv.append(hca)
+
+            # Extract values
+            hca["project.json"] = {
+                "uuid": str(uuid.uuid4()),
+                "title": obj["map"].get("Publication Title", "unknown"),
+                "id": obj["name"]
+            }
+            hca["sample.json"] = {
+                "uuid": str(uuid.uuid4()),
+                "donor": {"species": obj["map"]["Organism"]},
+            }
+            hca["assay.json"] = {
+                "imaging": {
+                    "microscope": "load from OME-XML",
+                },
+                "file": [
+                    {
+                        "format": "load-from-OMERO",
+                        "name": "load-from-OMERO",
+                    },
+                ]
+            }
+        print json.dumps(rv, indent=4, sort_keys=True)
+
+
+class TextPrinter(object):
```

Comment:
As mentioned separately, the object of this `TextPrinter was some initial console debugging while working on the study parser utility. Unless someone has a concrete need for it, I would not object to dropping it completely in favor of the generic JSON output.

Reply:
Fine by me. @dominikl, @pwalczysko ?